packages/arb-bridge-peripherals/contracts/tokenbridge/arbitrum/gateway/L2ArbitrumGateway.sol

```diff
@@ -243,14 +251,10 @@ abstract contract L2ArbitrumGateway is L2ArbitrumMessenger, TokenGateway {
         uint256 _amount,
         bytes calldata _data
     ) external payable override onlyCounterpartGateway {
-        (bytes memory gatewayData, bytes memory callHookData) = GatewayMessageHandler
+        (bytes memory gatewayData, bytes memory extraData) = GatewayMessageHandler
             .parseFromL1GatewayMsg(_data);
-
-        if (callHookData.length != 0) {
-            // callHookData should always be 0 since inboundEscrowAndCall is disabled
-            callHookData = bytes("");
-        }
-
+        // user encoded
+        (uint256 callHookGas, bytes memory callHookData) = abi.decode(extraData, (uint256, bytes));
```

Comment:
Let's instead handle this in the GatewayMessageHandler

Reply:
done in d8c24ac